src/integrationTest/java/org/opensearch/security/privileges/dlsfls/FlsFmIntegrationTests.java

```diff
@@ -0,0 +1,890 @@
+                                    || d.attrKeywordDocValuesDisabled().startsWith("dept_b")
+                                    || d.attrKeywordDocValuesDisabled().startsWith("dept_c")
+                            ).applyTransform(user.reference(DOC_WITH_FLS_FM_APPLIED))
+                        )
+                    )
+                );
+            } else {
+                assertThat(response, hasSearchHits(emptyHits()));
+            }
+        }
+    }
+
+    @Test
+    public void search_abilityToSearch_numericRange() {
+        try (TestRestClient client = cluster.getRestClient(user)) {
+            TestRestClient.HttpResponse response = client.get(
+                TEST_INDEX.name() + "/_search?q=" + URLEncoder.encode("attr_int:[5000 TO 9000]", StandardCharsets.US_ASCII)
+            );
+            assertThat(response, isOk());
+            if (user.reference(FIELD_IS_SEARCHABLE).test("attr_int")) {
+                assertThat(
+                    response,
+                    hasSearchHits(
+                        whereDocumentSourceEquals(
+                            TEST_DOCUMENTS.where(d -> d.attrInt() >= 5000 && d.attrInt() <= 9000)
+                                .applyTransform(user.reference(DOC_WITH_FLS_FM_APPLIED))
+                        )
+                    )
+                );
+            } else {
+                assertThat(response, hasSearchHits(emptyHits()));
+            }
+        }
+    }
+
+    /**
+     * The exists query internally operates on the _field_names field which gets special treatment in DlsFlsFilterLeafReader
+     */
+    @Test
+    public void search_abilityToSearch_exists() {
+        try (TestRestClient client = cluster.getRestClient(user)) {
+            TestRestClient.HttpResponse response = client.postJson(TEST_INDEX.name() + "/_search", """
+                {
+                  "query": {
+                    "exists": {
+                      "field": "attr_text_doc_values_disabled_nullable"
+                    }
+                  }
+                }""");
+            assertThat(response, isOk());
+            if (user.reference(FIELD_IS_SEARCHABLE).test("attr_text_doc_values_disabled_nullable")) {
+                assertThat(
+                    response,
+                    hasSearchHits(
+                        whereDocumentSourceEquals(
+                            TEST_DOCUMENTS.where(d -> d.content().containsKey("attr_text_doc_values_disabled_nullable"))
+                                .applyTransform(user.reference(DOC_WITH_FLS_FM_APPLIED))
+                        )
+                    )
+                );
+            } else {
+                assertThat(response, hasSearchHits(emptyHits()));
+            }
+        }
+    }
+
+    @Test
+    public void search_sortBy_explicitKeywordAttribute() {
+        if (user == Users.MASKING_ON_KEYWORD) {
+            // Sorting by a masked field produces an error 500. Skip this until this is fixed.
```

Comment:
Is there an open issue for this?

Reply:
no, that is also on my overflowing todo list :smiling_face_with_tear:

Reply:
reported issue